mach_eval/analyzers/electromagnetic/flux_linkage_analyzer.py

```diff
@@ -0,0 +1,141 @@
+import os
+import pandas as pd
+import numpy as np
+from time import time as clock_time
+
+class FluxLinkageJMAG_Problem:
+    """Problem class for processing FEA flux linkages
+    Attributes:
+        toolJmag: JMAG tool required for FEA scripting
+        phase_names: names of phases in machine
+        rated_current: rated current of machine
+    """
+
+    def __init__(self, toolJmag, phase_names, rated_current):
+        self.toolJmag = toolJmag
+        self.phase_names = phase_names
+        self.rated_current = rated_current
+    
+    def add_em_study(self, study_name):
+
+        self.app = self.toolJmag.jd
+        self.model = self.app.GetCurrentModel()
+
+        # Pre-processing
+        self.model.SetName("Machine_FluxLinkage_Project")
+
+        # Initialize JMAG application
+        self.app.SetCurrentStudy(study_name)
+        study = self.model.GetStudy(study_name)
+
+        # Set csv folder output
+        self.results_filepath = os.path.dirname(__file__) + "/run_data/"
+
+        # Create output folder
+        if not os.path.isdir(self.results_filepath):
+            os.makedirs(self.results_filepath)
+
+        # Create csv output file loction
+        study.GetStudyProperties().SetValue("CsvOutputPath", self.results_filepath)
+        study.GetStudyProperties().SetValue("CsvResultTypes", "FEMCoilFlux")
+
+        self.app.SetCurrentStudy(study_name)
+
+        return study
+
+    
+    def del_op_pts(self, study):
```

Comment:
Let's rename to `delete_operating_points(self)` 
--> and no need to bring `study` in